WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -617,6 +617,45 @@
         expect(charge.metadata.competition).to eq competition.name
       end
 
+      it 'processes payment with donation and valid credit card' do
```

Comment:
One approach is to test the basic function fully for one case, and only test the code that is different fore the second.

Personally I'd factor out a token function with default parameters right away.

Also: `Time.now.year+1`